OneCommunityGlobal/HighestGoodNetworkApp
# Luis summary intro btn

# Description
Added a copy button to the user's profile, which copies and generates the summary intro for managers, assistant managers and administrators.
Copies the text to the user's clipboard, and checks if the user's in the team have submitted their weekly summaries, and categorizes based on that and saves it to the clipboard.

![](https://user-images.githubusercontent.com/22454953/274789706-8c84e40c-54f4-4376-bc62-8f5575f4c910.png)




## Main changes explained:
Added a new button to the user's profile which generates the string to the clipboard.
Also, utilized existing permissions, using hasPermission method. Since the user's I am targeting are Managers, Admins and Owners. They all have getProjectMembers permission, so I am using this permission to check if they can see the button.
 
Only Managers, Admins and Owners should be able to see the button. 
**Volunteers cannot see the button.**

## How to test:
1. check into current branch
2. do npm install and ... to run this PR locally
3. log as admin/owner
4. go to dashboard→ View profile
5. Check if "Generate Summary Intro" appears
6. Click the button, and paste what appears.
7. Check if the user's who submitted a summary have and those who haven't are listed in the correct spot.
8. Make sure your user is a apart of a team, to view all the correct members who have and who haven't submitted a summary.
9. The current user is assigned the manager role since they are clicking the button.
10. Retest using a volunteer account and the button should not appear.


## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/3e2dee14-f4c0-45c1-a55a-cf7ed372a7fb


## Note:
If you have a suggestion to improve the code that would be great. 
Thanks. 
The following warnings appear plaese disregard them 
<img width="324" alt="Screenshot 2023-10-18 at 8 40 42 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/598844b2-3c49-4e5a-b00f-61dc3d3e7567">
